hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractTableDetailsLayout.java

```diff
@@ -156,44 +192,29 @@ protected void setName(final String headerCaption, final String value) {
 
     private void restoreState() {
         if (onLoadIsTableRowSelected()) {
-            populateData(true);
+            populateData(null);
+            editButton.setEnabled(true);
         }
         if (onLoadIsTableMaximized()) {
-            /**
-             * If table is maximized hide details layout.
-             */
-            hideLayout();
+            setVisible(false);
         }
     }
 
-    protected void showLayout() {
-        setVisible(true);
-    }
-
-    protected void hideLayout() {
-        setVisible(false);
-    }
-
     /**
      * If no data in table (i,e no row selected),then disable the edit button.
      * If row is selected ,enable edit button.
      */
-    protected void populateData(final Boolean isRowSelected) {
-        if (isRowSelected) {
-            populateDetailsWidget();
-            enableEditButton();
+    private void populateData(final T selectedBaseEntity) {
+        this.selectedBaseEntity = selectedBaseEntity;
+        editButton.setEnabled(selectedBaseEntity != null);
```

Comment:
Looks kind of curious to me, isn't it better to have a parameter before that determines a boolean with that statement?